clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/ProjectController.groovy

```diff
@@ -28,29 +29,45 @@ import org.springframework.web.bind.annotation.RequestMapping
 import org.springframework.web.bind.annotation.RequestMethod
 import org.springframework.web.bind.annotation.RestController
 
+import static com.netflix.spinnaker.clouddriver.core.ProjectClustersService.ClusterModel
+
 @Slf4j
 @RestController
 @RequestMapping("/projects/{project}")
 class ProjectController {
 
   Cache cacheView
+  ProjectClustersService projectClustersService
+  ProjectClustersCachingAgentProperties projectClustersCachingAgentProperties
 
   @Autowired
-  ProjectController(Cache cacheView) {
+  ProjectController(Cache cacheView,
+                    ProjectClustersService projectClustersService,
+                    ProjectClustersCachingAgentProperties projectClustersCachingAgentProperties) {
     this.cacheView = cacheView
+    this.projectClustersService = projectClustersService
+    this.projectClustersCachingAgentProperties = projectClustersCachingAgentProperties
   }
 
   @RequestMapping(method= RequestMethod.GET, value = "/clusters")
-  List<ProjectClustersCachingAgent.ClusterModel> getClusters(@PathVariable String project) {
-    CacheData cacheData = cacheView.get(Namespace.PROJECT_CLUSTERS.ns, "v1")
-    if (cacheData == null) {
-      throw new NotFoundException("Projects not cached")
-    }
+  List<ClusterModel> getClusters(@PathVariable String project) {
+    if (projectClustersCachingAgentProperties.allowList.contains(project)) {
```

Comment:
not sure if there's a problem with possible case-sensitivity here but otherwise lgtm.

Reply:
Will change. Just spot-checked the front50 API and that's case insensitive. Was erring on the side of caution here, but looks like that's not necessary here.